sqlmesh/schedulers/airflow/plan.py

```diff
@@ -2,16 +2,86 @@
 
 import typing as t
 
+import pandas as pd
+from sqlglot import exp
+
 from sqlmesh.core import scheduler
+from sqlmesh.core.engine_adapter import EngineAdapter
 from sqlmesh.core.environment import Environment
 from sqlmesh.core.plan import can_evaluate_before_promote
 from sqlmesh.core.snapshot import SnapshotTableInfo
-from sqlmesh.core.state_sync import StateSync
+from sqlmesh.core.state_sync import EngineAdapterStateSync, StateSync
+from sqlmesh.core.state_sync.base import DelegatingStateSync
 from sqlmesh.schedulers.airflow import common
+from sqlmesh.utils import nullsafe_join
 from sqlmesh.utils.date import now, now_timestamp
 from sqlmesh.utils.errors import SQLMeshError
 
 
+class PlanDagState:
+    def __init__(self, engine_adapter: EngineAdapter, schema: t.Optional[str]):
```

Comment:
add a catalog here too cc @eakmanrq

Reply:
we currently don't support it for `StateSync` either.